interfaces/builtin/docker_support.go

```diff
@@ -185,156 +185,20 @@ pivot_root,
-change_profile unsafe /snap/core/*/usr/lib/snapd/snap-conf[^i]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/core/*/usr/lib/snapd/snap-confi[^n]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/core/*/usr/lib/snapd/snap-confin[^e]** -> cri-containerd.apparmor.d,
-
-# branch for the /snap/snapd/... paths
-change_profile unsafe /snap/snap[^d]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd[^/]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/[^u]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/u[^s]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/us[^r]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr[^/]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/[^l]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/l[^i]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/li[^b]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib[^/]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/[^s]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/s[^n]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/sn[^a]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/sna[^p]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snap[^d]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd[^/]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/[^s]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/s[^n]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/sn[^a]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/sna[^p]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap[^-]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-[^c]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-c[^o]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-co[^n]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-con[^f]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-conf[^i]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-confi[^n]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-confin[^e]** -> cri-containerd.apparmor.d,
+# below are auto-generated rules using GenerateAAREExclusionPatterns
+###EXCL{change_profile unsafe <> -> cri-containerd.apparmor.d,:/snap/snapd/*/usr/lib/snapd/snap-confine,/snap/core/*/usr/lib/snapd/snap-confine}###
```

Comment:
The patters, in not expanded, would be treated as valid comments and also their format is non trivial. We have a check for ensuring that a call to InsertAARE.. fails when the placeholder was not found, but we have no checks for the opposite case, i.e. someone adds a pattern, but there is no expansion for it. 

Now, we seem to have the assumptions that placeholders generally follow a pattern of `###.*###`, so I'm wondering if adding such check somewhere in Backend.addContent where we deal with `###SNIPPETS###` could be useful.

Maybe a followup? This PR is already quite large.